### PR TITLE
main-2.0: add new URL for SAS 9.4 M8

### PR DESCRIPTION
**Pull request by bryanpaget**
This update is to remove the SAS image's dependency on sas4c, which is the older SAS image using SAS 9.4 M7.

I have uploaded SAS 9.4 M8 to an Azure storage container and moved the relevant logic out of sas4c into the AAW SAS image.

Todo:
- [x] Add secrets for storage account to Github Actions.

Ticket: https://jirab.statcan.ca/browse/BTIS-304

### CODE REVIEW

File: docker-bits/6_sas.Dockerfile [hunk 1]
```diff
@@ -11,22 +11,39 @@ RUN wget -q ${QUARTO_URL} -O /tmp/quarto-${QUARTO_VERSION}-linux-amd64.tar.gz &&
     chmod +x /tmp/quarto-${QUARTO_VERSION} && \
     ln -s /tmp/quarto-${QUARTO_VERSION}/bin/quarto /usr/bin/quarto
 
-RUN groupadd -g 1002 sasstaff && \
+RUN groupadd -g 1337 supergroup && \
```
Comment (mathis-marcotte) [inline docker-bits/6_sas.Dockerfile:14]: This is reverting a PR I made. Is this actually necessary?

File: docker-bits/6_jupyterlab.Dockerfile [hunk 2]
```diff
@@ -47,7 +47,7 @@ COPY vscode-overrides.json $CS_TEMP_HOME/Machine/settings.json
 # Languagepacks.json needs to exist for code-server to recognize the languagepack
 COPY languagepacks.json $CS_TEMP_HOME/
 
-RUN pip install \
+RUN pip install --no-cache-dir --quiet \
```
Comment (mathis-marcotte) [inline docker-bits/6_jupyterlab.Dockerfile:50]: Is this necessary? It seems irrelevant to the PR

**Review by mathis-marcotte — CHANGES_REQUESTED** [review]

File: docker-bits/6_sas.Dockerfile [hunk 3]
```diff
@@ -11,22 +11,39 @@ RUN wget -q ${QUARTO_URL} -O /tmp/quarto-${QUARTO_VERSION}-linux-amd64.tar.gz &&
...
 
 RUN apt-get update && apt-get install -y --no-install-recommends \
     libmagic1 \
     && rm -rf /var/lib/apt/lists/*
 
-RUN ln -s /usr/local/SASHome/SASFoundation/9.4/bin/sas_en /usr/local/bin/sas && \
-    chmod -R 0775 /usr/local/SASHome/studioconfig
-
-WORKDIR /home/jovyan
+WORKDIR /home/sas
```
Comment (mathis-marcotte) [inline docker-bits/6_sas.Dockerfile:46]: We don't have a sas user, so we shouldnt have a home/sas

**Review by mathis-marcotte — CHANGES_REQUESTED** [review]

File: docker-bits/6_jupyterlab.Dockerfile (hunk 2 above)
Comment (bryanpaget) [reply docker-bits/6_jupyterlab.Dockerfile]: I think we need this

File: docker-bits/6_sas.Dockerfile (hunk 3 above)
Comment (bryanpaget) [reply docker-bits/6_sas.Dockerfile]: Good catch, thank you. It was something leftover from the sas4c image.

File: docker-bits/6_sas.Dockerfile (hunk 1 above)
Comment (bryanpaget) [reply docker-bits/6_sas.Dockerfile]: It's just an old branch, I opened the branch before you made those changes, so this branch still had the old stuff. I've made the changes to match the work you did.

File: docker-bits/6_jupyterlab.Dockerfile (hunk 2 above)
Comment (bryanpaget) [reply docker-bits/6_jupyterlab.Dockerfile]: I think the --no-cache-dir and --quiet flags are necessary but I'm building now without them to see what happens. We tend to use those flags throughout the dockerfiles but not consistently. I remember having a permission error related to the cache dir and when I looked at line 50 I noticed it didn't have the --no-cache-dir flag, so I decided to add it along with --quiet for consistency with other calls to pip install.

Comment (mathis-marcotte) [reply docker-bits/6_jupyterlab.Dockerfile]: Ok, good enough for me. We can keep those in, just wanted some clarification, thanks!

File: docker-bits/6_sas.Dockerfile [hunk 4]
```diff
@@ -15,19 +15,30 @@ RUN groupadd -g 1002 sasstaff && \
...
 COPY --from=minio/mc:RELEASE.2022-03-17T20-25-06Z /bin/mc /usr/local/bin/mc
 
 RUN apt-get update && apt-get install -y --no-install-recommends \
     libmagic1 \
     && rm -rf /var/lib/apt/lists/*
 
-RUN ln -s /usr/local/SASHome/SASFoundation/9.4/bin/sas_en /usr/local/bin/sas && \
-    chmod -R 0775 /usr/local/SASHome/studioconfig
-
-WORKDIR /home/jovyan
```
Comment (mathis-marcotte) [inline docker-bits/6_sas.Dockerfile:29]: Removing the workdir impacts the rest of the dockerfile, mainly the RUN and COPY commands, is this actually a needed change?

**Review by mathis-marcotte — CHANGES_REQUESTED** [review]

File: docker-bits/6_sas.Dockerfile (hunk 1 above)
Comment (bryanpaget) [reply docker-bits/6_sas.Dockerfile]: There was another user/permissions error:

Line 33:
``` bash
chown -R sas:sasstaff /usr/local/SASHome --> chown -R jovyan:sasstaff /usr/local/SASHome
```

File: docker-bits/6_jupyterlab.Dockerfile (hunk 2 above)
Comment (bryanpaget) [reply docker-bits/6_jupyterlab.Dockerfile]: This is the error I was getting. 
``` bash
WARNING: The directory '/home/jovyan/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
```
Context:
``` bash
Step 47/85 : RUN pip install     'git+https://github.com/betatim/vscode-binder' &&     mamba install --quiet --yes -c conda-forge     'pkg-config'     'jupyterlab'     'jupyter_contrib_nbextensions'     'dash'     'plotly'     'ipywidgets'     'markupsafe'     'ipympl'     'pexpect==4.9.0'     'jupyter-resource-usage'     'jupyter-server-proxy==4.2.0'     'jupyterlab-language-pack-fr-fr'     'jupyterlab_execute_time'     'nb_conda_kernels'     'jupyterlab-lsp'     'jupyter-lsp'  &&     jupyter server extension enable --py jupyter_server_proxy &&     jupyter nbextension enable codefolding/main --sys-prefix &&     jupyter labextension enable       '@jupyterlab/translation-extension'       '@jupyterlab/server-proxy'       'nbdime-jupyterlab'     &&     jupyter lab build &&     jupyter lab clean &&   clean-layer.sh &&   rm -rf /home/$NB_USER/.cache/yarn &&   rm -rf /home/$NB_USER/.node-gyp &&   fix-permissions $CONDA_DIR &&   fix-permissions /home/$NB_USER
 ---> Running in dbcc9a6ffc88
WARNING: The directory '/home/jovyan/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
```

File: docker-bits/6_sas.Dockerfile (hunk 4 above)
Comment (bryanpaget) [reply docker-bits/6_sas.Dockerfile:29]: I'll add it back.

**Review by mathis-marcotte — CHANGES_REQUESTED** [review]
Tried in a notebook on dev and I can't open SAS Studio.
image

image

**Pull request closed without merging**